### PR TITLE
[CS-3127] Update PaymentRequest link share info

### PR DESCRIPTION
**Pull request by topwebtek7**
### Description

Updated PaymentRequest link share info.

- [x] Completes #CS-3127

### Checklist

- [x] All UI changes have been tested on a small device

### Screenshots

Screen Shot 2022-02-23 at 9 56 56 PM
Screen Shot 2022-02-23 at 9 52 20 PM

### CODE REVIEW

**Review by oidoug — APPROVED** [review]
LGTM!

File: cardstack/src/utils/merchant-utils.ts [hunk 1]
```diff
@@ -114,23 +114,30 @@ export const shareRequestPaymentLink = (
...
+      };
 
-  return Share.share(
-    {
-      message,
-      title,
-    },
-    options
-  );
+  return Share.share(content, options);
```
Comment (danibonilha) [inline cardstack/src/utils/merchant-utils.ts]: nit: a way to minimize checking the Device, is to use Platform.select, so we could replace this piece with: 
```ts


  const shareConfig = {
    options: Platform.select({
      android: {
        dialogTitle: title,
      },
      ios: {
        excludedActivityTypes,
        subject: title,
      },
    }),
    content: Platform.select({
      android: {
        title,
        message,
      },
      ios: {
        title,
        message,
        url: paymentRequestLink,
      },
      default: { message },
    }),
  };

  return Share.share(shareConfig.content, shareConfig.options);
```

Comment (danibonilha) [reply cardstack/src/utils/merchant-utils.ts]: Then on the tests we need to mock the platform select: 

```ts
  describe('shareRequestPaymentLink', () => {
    beforeAll(() => {
      jest
        .spyOn(Platform, 'select')
        .mockImplementation(obj => obj[Platform.OS]);
    });
```
And  replace  
- `Device.isIOS = true` with `Platform.OS = 'ios' `
- `Device.isIOS = false` with `Platform.OS = 'android' `

**Review by danibonilha — COMMENTED** [review]
It's looking good, just a small suggestion to "shrinking" it, but it's nit

**Pull request merged**